rotkehlchen/chain/base/modules/basenames/decoder.py

```diff
@@ -0,0 +1,224 @@
+        if node == BASE_ETH_NODE:
+            return 'base.eth'
+        elif node == BASE_REVERSE_NODE:
+            return 'reverse'
+
+        # Try to retrieve it from the cache
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            if (name_to_show := globaldb_get_unique_cache_value(
+                    cursor=cursor,
+                    key_parts=(CacheType.ENS_NAMEHASH, node.hex()),
+            )) is not None:
+                return name_to_show
+
+        # Call the L2 Resolver contract's name method.
+        try:
+            if not (name_to_show := self.evm_inquirer.contracts.contract(BASENAMES_L2_RESOLVER).call(  # noqa: E501
+                node_inquirer=self.evm_inquirer,
+                method_name='name',
+                arguments=[node],
+            )):
+                log.error(f'Blank name returned from resolver contract for node {node.hex()}.')
+                return None
+        except RemoteError as e:
+            log.error(f'Failed to find full Basename for node {node.hex()} due to {e!s}')
+            return None
+
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            globaldb_set_unique_cache_value(
+                write_cursor=write_cursor,
+                key_parts=(CacheType.ENS_NAMEHASH, node.hex()),
+                value=name_to_show,
+            )
+        return name_to_show
+
+    def _maybe_get_labelhash_name(
+            self,
+            context: DecoderContext,
+            label_hash: str,
+            node: bytes | None = None,
+    ) -> str | None:
+        """Try to get the name corresponding to a labelhash.
+        Checks the cache first and then queries the base.org api.
+        Returns the name or None if unable to retrieve a name.
+        """
+        label_hash = label_hash.removeprefix('0x')
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            if (found_name := globaldb_get_unique_cache_value(
+                cursor=cursor,
+                key_parts=(CacheType.ENS_LABELHASH, label_hash),
+            )):
+                return found_name
+
+        try:
+            response_data = requests.get(
+                url=f'https://www.base.org/api/basenames/metadata/{int(label_hash, 16)}',
+                timeout=CachedSettings().get_timeout_tuple(),
+            ).json()
+        except (requests.exceptions.RequestException, JSONDecodeError, ValueError) as e:
+            log.error(
+                f'Failed to retrieve name for labelhash {label_hash} '
+                f'from Basenames api due to {e!s}',
+            )
+            return None
+
+        try:
+            with GlobalDBHandler().conn.write_ctx() as write_cursor:
+                globaldb_set_unique_cache_value(
+                    write_cursor=write_cursor,
+                    key_parts=(CacheType.ENS_LABELHASH, label_hash),
+                    value=(name := response_data['name']),
+                )
+        except (TypeError, KeyError):
```

Comment:
So this is not giving enough information in the logs.

If what you want to check here is that `response_data` is a dict then you should explicitly check for that. There is no way to know when you read the code why the type error is checked for. Check for it above.

Also then this just leaves key error in which case you should simply mention it's a missing attribute (and which one it is) from the response.